website/src/components/ColorGenerator/index.tsx

```diff
@@ -5,117 +5,192 @@
+import React, {useEffect, useState} from 'react';
+import clsx from 'clsx';
 import Color from 'color';
 import CodeBlock from '@theme/CodeBlock';
+import Admonition from '@theme/Admonition';
+import Link from '@docusaurus/Link';
+import {useColorMode} from '@docusaurus/theme-common';
 
+import {
+  type ColorState,
+  COLOR_SHADES,
+  LIGHT_PRIMARY_COLOR,
+  DARK_PRIMARY_COLOR,
+  LIGHT_BACKGROUND_COLOR,
+  DARK_BACKGROUND_COLOR,
+  getAdjustedColors,
+  lightStorage,
+  darkStorage,
+} from '@site/src/utils/colorUtils';
 import styles from './styles.module.css';
 
-const COLOR_SHADES: Record<
-  string,
-  {
-    adjustment: number;
-    adjustmentInput: string;
-    displayOrder: number;
-    codeOrder: number;
-  }
-> = {
-  '--ifm-color-primary': {
-    adjustment: 0,
-    adjustmentInput: '0',
-    displayOrder: 3,
-    codeOrder: 0,
-  },
-  '--ifm-color-primary-dark': {
-    adjustment: 0.1,
-    adjustmentInput: '10',
-    displayOrder: 4,
-    codeOrder: 1,
-  },
-  '--ifm-color-primary-darker': {
-    adjustment: 0.15,
-    adjustmentInput: '15',
-    displayOrder: 5,
-    codeOrder: 2,
-  },
-  '--ifm-color-primary-darkest': {
-    adjustment: 0.3,
-    adjustmentInput: '30',
-    displayOrder: 6,
-    codeOrder: 3,
-  },
-  '--ifm-color-primary-light': {
-    adjustment: -0.1,
-    adjustmentInput: '-10',
-    displayOrder: 2,
-    codeOrder: 4,
-  },
-  '--ifm-color-primary-lighter': {
-    adjustment: -0.15,
-    adjustmentInput: '-15',
-    displayOrder: 1,
-    codeOrder: 5,
-  },
-  '--ifm-color-primary-lightest': {
-    adjustment: -0.3,
-    adjustmentInput: '-30',
-    displayOrder: 0,
-    codeOrder: 6,
-  },
-};
-
-const DEFAULT_PRIMARY_COLOR = '3578e5';
+function wcagContrast(foreground: string, background: string) {
+  const contrast = Color(foreground).contrast(Color(background));
+  // eslint-disable-next-line no-nested-ternary
+  return contrast > 7 ? 'AAA 🏅' : contrast > 4.5 ? 'AA 👍' : 'Fail 🔴';
+}
 
 function ColorGenerator(): JSX.Element {
+  const {isDarkTheme, setDarkTheme, setLightTheme} = useColorMode();
+  const DEFAULT_PRIMARY_COLOR = isDarkTheme
+    ? DARK_PRIMARY_COLOR
+    : LIGHT_PRIMARY_COLOR;
+  const DEFAULT_BACKGROUND_COLOR = isDarkTheme
+    ? DARK_BACKGROUND_COLOR
+    : LIGHT_BACKGROUND_COLOR;
+
+  const [inputColor, setInputColor] = useState(DEFAULT_PRIMARY_COLOR);
   const [baseColor, setBaseColor] = useState(DEFAULT_PRIMARY_COLOR);
+  const [background, setBackground] = useState(DEFAULT_BACKGROUND_COLOR);
   const [shades, setShades] = useState(COLOR_SHADES);
-  const color = Color(`#${baseColor}`);
-  const adjustedColors = Object.keys(shades)
-    .map((shade) => ({
-      ...shades[shade],
-      variableName: shade,
-    }))
-    .map((value) => ({
-      ...value,
-      hex: color.darken(value.adjustment).hex(),
-    }));
+  const [storage, setStorage] = useState(
+    isDarkTheme ? darkStorage : lightStorage,
+  );
+
+  // init -> create default storage values
+  useEffect(() => {
```

Comment:
probably overkill to create those storage values on mount: we could wait for user to interact with the color generator